### PR TITLE
ADEN-13176 - Stickiness on N&R

### PR DESCRIPTION
**Pull request by adi-wesolowski-fandom**
ADEN-13176 - stickiness on N&R:

- apply sticky template to all n&r sites, enabling this template as default, including SPA
- it is not active until lineitem show up in icbm variable
- default behaviour can be still overwritten by "force" icbm variable as on fandom site

Jira task: https://fandom.atlassian.net/browse/ADEN-13176

source branch: ADEN-13135 (!)

### CODE REVIEW

**Review by dmnsobczak — APPROVED** [review]

File: src/platforms/shared/templates/handlers/sticky-tlb/sticky-tlb-blocking-handler.ts [hunk 1]
```diff
@@ -45,19 +45,27 @@ export class StickyTlbBlockingHandler implements TemplateStateHandler {
...
+	}
 
-					return (
-						lineId.toString() === this.adSlot.lineItemId.toString() &&
-						(!geo || utils.geoService.isProperGeo([geo]))
-					);
-				});
+	private checkIdsMapBySlotGamId(gamId: string, map: string[]): boolean {
+		if (!Array.isArray(map)) {
+			return true;
```
Comment (dmnsobczak) [inline src/platforms/shared/templates/handlers/sticky-tlb/sticky-tlb-blocking-handler.ts:59]: I suggest `false` here. This check here is to prevent AdEngine from crashing when somebody will break icVariable. In that case it's better to disable sticking rather than enable it globally 🤔 
On the other hand, ICBM have syntax checking now, so this might be redundant now.

File: src/platforms/shared/templates/handlers/sticky-tlb/sticky-tlb-blocking-handler.ts [hunk 2]
```diff
@@ -45,19 +45,27 @@ export class StickyTlbBlockingHandler implements TemplateStateHandler {
...
 		}
-		return false;
+
+		return map
+			.map((element) => element.toString())
+			.some((element) => {
+				const [mapGamId, geo] = element.split(':', 2);
```
Comment (dmnsobczak) [inline src/platforms/shared/templates/handlers/sticky-tlb/sticky-tlb-blocking-handler.ts:65]: Nobody is using this "per geo feature" :( But whatever.

**Review by dmnsobczak — DISMISSED** [review]
@adi-wesolowski-fandom One suggestion: `true` => `false`

File: src/platforms/shared/templates/handlers/sticky-tlb/sticky-tlb-blocking-handler.ts (hunk 2 above)
Comment (adi-wesolowski-fandom) [reply src/platforms/shared/templates/handlers/sticky-tlb/sticky-tlb-blocking-handler.ts:65]: I wasnt sure. It is here, so I thought,maybe somewhere it is. And removing it, change logic, and need to tell business, we simplified it. I can try to simplify it if you insist :)

File: src/platforms/shared/templates/handlers/sticky-tlb/sticky-tlb-blocking-handler.ts (hunk 1 above)
Comment (adi-wesolowski-fandom) [reply src/platforms/shared/templates/handlers/sticky-tlb/sticky-tlb-blocking-handler.ts:59]: ok

File: src/platforms/shared/templates/handlers/sticky-tlb/sticky-tlb-blocking-handler.ts (hunk 2 above)
Comment (dmnsobczak) [reply src/platforms/shared/templates/handlers/sticky-tlb/sticky-tlb-blocking-handler.ts:65]: Noo, it's fine :)

Comment (adi-wesolowski-fandom) [reply src/platforms/shared/templates/handlers/sticky-tlb/sticky-tlb-blocking-handler.ts:65]: done here: https://github.com/Wikia/ad-engine/pull/2029

File: src/platforms/shared/templates/handlers/sticky-tlb/sticky-tlb-blocking-handler.ts [hunk 3]
```diff
@@ -56,7 +56,7 @@ export class StickyTlbBlockingHandler implements TemplateStateHandler {
 
 	private checkIdsMapBySlotGamId(gamId: string, map: string[]): boolean {
 		if (!Array.isArray(map)) {
-			return true;
```
Comment (preludek) [inline src/platforms/shared/templates/handlers/sticky-tlb/sticky-tlb-blocking-handler.ts]: This template is used between all platforms, is that change necessary and safe? I see potential regression.

Also if it's safe we can change it to a ternary in return `return Array.isArray(map) ? map : false` something like that

File: src/platforms/shared/templates/handlers/sticky-tlb/sticky-tlb-blocking-handler.ts [hunk 4]
```diff
@@ -56,7 +56,7 @@ export class StickyTlbBlockingHandler implements TemplateStateHandler {
...
+			return false;
```
Comment (preludek) [inline src/platforms/shared/templates/handlers/sticky-tlb/sticky-tlb-blocking-handler.ts:59]: This template is used between all platforms, is that change necessary and safe? I see potential regression.

Also if it's safe we can change it to a ternary in return `return Array.isArray(map) ? map : false` something like that

File: src/platforms/news-and-ratings/shared/styles/uap-styles.scss [hunk 5]
```diff
@@ -17,7 +17,7 @@
 
 			@include bfaa-slot();
 			@include uap-porvata();
-			@include sticky-tlb-slot();
+			@include sticky-tlb-slot(0px, 5000061);
```
Comment (preludek) [inline src/platforms/news-and-ratings/shared/styles/uap-styles.scss:20]: I kinda dislike that. Wa are using a random high z-index here and that can bite us in the ass in the future. If it's necessary to change it please move it to some sort of a variable:
1. to keep all of those in one place
2. to have it named and to know right away what it's for

File: src/platforms/shared/templates/handlers/sticky-tlb/sticky-tlb-blocking-handler.ts (hunk 4 above)
Comment (preludek) [reply src/platforms/shared/templates/handlers/sticky-tlb/sticky-tlb-blocking-handler.ts:59]: Ok, I got the answer for first question here - https://github.com/Wikia/ad-engine/pull/2018/files/990fe7512e1ad18093bd6ff47053a7ee188f391c#r1250580706 :)

Comment (adi-wesolowski-fandom) [reply src/platforms/shared/templates/handlers/sticky-tlb/sticky-tlb-blocking-handler.ts:59]: ok

File: src/platforms/news-and-ratings/shared/styles/uap-styles.scss (hunk 5 above)
Comment (adi-wesolowski-fandom) [reply src/platforms/news-and-ratings/shared/styles/uap-styles.scss:20]: its the highest across N&R I found. I put it here as it has visible impact. But moving it to ICBM is a good option, but - wouldnt be too much to put such a simple thing there? MAybe you are right. We can do that.

Comment (adi-wesolowski-fandom) [reply src/platforms/news-and-ratings/shared/styles/uap-styles.scss]: moved it to configuration per each site separately. should be more clear now

**Review by preludek — APPROVED** [review]

**Review by kwfandom — APPROVED** [review]
